src/components/networks/utils.js

```diff
@@ -101,28 +86,36 @@ export function validateNetmask(prefixOrNetmask) {
  * @returns {string}
  */
 export function netmaskConvert(prefixOrNetmask) {
-    const parts = prefixOrNetmask.split('.');
+    // single number → netmask
+    if (/^[0-9]+$/.test(prefixOrNetmask)) {
+        try {
+            return ipaddr.IPv4.subnetMaskFromPrefixLength(parseInt(prefixOrNetmask)).toString();
+        } catch (_ex) {
+            // leave unchanged; UI will validate
+        }
+    }
 
-    if (parts.length === 4)
-        return prefixOrNetmask;
-    else if (parts.length === 1)
-        return ip.fromPrefixLen(prefixOrNetmask);
+    return prefixOrNetmask;
 }
 
 /**
- * Checks whetever address @ip is in subnet defined by @network and @netmask
+ * Checks whetever address @address is in subnet defined by @network and @netmask
  *
  * @param {string} network
  * @param {string} netmask
- * @param {string} ip
+ * @param {string} address
  * @returns {boolean}
  */
-export function isIpv4InNetwork(network, netmask, ipaddr) {
-    if (!ip.isV4Format(network) || !validateNetmask(netmask) || !ip.isV4Format(ipaddr))
+export function isIpv4InNetwork(network, netmask, address) {
+    if (!validateIpv4(network) || !validateIpv4(address))
+        return false;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1455-20240222-112249-96a33f93-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)